test/nullable.jl

```diff
@@ -279,3 +279,14 @@ end
 
 # issue #11675
 @test repr(Nullable()) == "Nullable{Union{}}()"
+
+# Ensure that promotion works
+types = [
+    (DateTime, DateTime),
+    (Float64, Int64),
+]
+
+for (S, T) in types
+    @test Base.promote_type(Nullable{S}, Nullable{T}) == Nullable{Base.promote_type(S, T)}
```

Comment:
Actually testing arithmetic operations isn't possible in Base of course until we define arithmetic operations there. Could you simply apply the changes I suggested in my first comment then? I think we can merge this soon since we all agree that it's needed.